pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -187,6 +188,31 @@ func remaining(t time.Time) time.Duration {
 	return t.Sub(time.Now()).Round(time.Second)
 }
 
+func dumpClusterAutoscalerLogs(client runtimeclient.Client, restClient *rest.RESTClient) {
+	pods := corev1.PodList{}
+	caLabels := map[string]string{
+		"app": "cluster-autoscaler",
+	}
+	if err := client.List(context.TODO(), &pods, []runtimeclient.ListOptionFunc{runtimeclient.MatchingLabels(caLabels)}...); err != nil {
+		glog.Errorf("Error querying api for clusterAutoscaler pod object: %v", err)
+	}
+	if len(pods.Items) > 1 {
+		glog.Errorf("Got more than one autoscaler pods, which is unexpected!!!")
+	}
+	// We're only expecting one pod but let's log from all that
+	// are found. If we see more than one that's indicative of
+	// some unexpected problem and we may as well dump its logs.
+	for _, pod := range pods.Items {
+		req := restClient.Get().Namespace(e2e.TestContext.MachineApiNamespace).Resource("pods").Name(pod.Name).SubResource("log")
```

Comment:
is this not supported on controller runtime yet?

Reply:
AFAIU, No!